db/crud.go

```diff
@@ -126,14 +127,14 @@ func (db *DatabaseContext) GetDocSyncData(docid string) (SyncData, error) {
 	if db.UseXattrs() {
 		// Retrieve doc and xattr from bucket, unmarshal only xattr.
 		// Triggers on-demand import when document xattr doesn't match cas.
-		var rawDoc, rawXattr []byte
-		cas, getErr := db.Bucket.GetWithXattr(key, base.SyncXattrName, &rawDoc, &rawXattr)
+		var rawDoc, rawXattr, rawUserXattr []byte
+		cas, getErr := db.Bucket.GetWithXattr(key, base.SyncXattrName, db.Options.UserXattrKey, &rawDoc, &rawXattr, &rawUserXattr)
```

Comment:
I don't think GetDocSyncData is used on any paths that write documents.  Since it's explicitly documented above as only retrieving the sync metadata, I think we should pass "" as UserXattrKey here to avoid unnecessary work.  It doesn't look like GetWithXattr is triggering any on-demand import, so that shouldn't be a factor.

Reply:
Yeah good point. Will do that.

Reply:
Actually upon reviewing it GetDocSyncData does trigger an on-demand import if required which does require us to have the user xattr available to us.